### DISCUSSION TITLE
Two questions on durable HTTP sink

### DISCUSSION THREAD

**Discussion by pankaj-k** [FantasticFiasco/serilog-sinks-http#198] [General]
Hi,

I will like to have some clarity on the durability option of this sink.

1. Will it always write to disk and then send it? Or it will write only when log server is not reached?
2. Will it delete the buffer files after they have been processed and data sent? Or will it still retain them till limit defined by retainedBufferFileCountLimit is breached and then it will delete the older buffer files?

**Comment by FantasticFiasco** [comment]
1. The sink will always first write the log events to disk, and then periodically send them over the network to the log server.
2. Buffer files where all log events have been sent will be deleted. If the log server is accepting the log events, at most 2 buffer files at any given time will be present on disk.

**Reply by pankaj-k** [reply]
Thanks for the clarification!! Saves me some experimentations !!!
